src/lib/server/websearch/markdown/tree.ts

```diff
@@ -0,0 +1,63 @@
+import type { SerializedHTMLElement } from "../scrape/types";
+import { htmlElementToMarkdownElements, mergeAdjacentElements } from "./fromHtml";
+import type { HeaderElement, MarkdownElement } from "./types";
+import { MarkdownElementType } from "./types";
+import { chunkElements } from "./utils/chunk";
+
+/**
+ * Converts HTML elements to Markdown elements and creates a tree based on header tags
+ * For example: h1 [h2 [p p blockquote] h2 [h3 [...] ] ]
+ **/
+export function htmlToMarkdownTree(
+	title: string,
+	htmlElements: SerializedHTMLElement[],
+	maxCharsPerElem: number
+): HeaderElement {
+	let parent: HeaderElement = {
```

Comment:
```suggestion
	let root: HeaderElement = {
```
I think `root` could be a better var name here